### PR TITLE
Check autostart after calling load at the end of current event loop

### PR DESCRIPTION
**Pull request by robwalch**
### This PR will...

Defer the `_checkAutoStart` call made after `load` on "itemReady" until after the current callstack is done executing. If `play` or `stop` is called, the deferred `_checkAutoStart` call is cancelled.

This is done by calling `autostartFallbackOnItemReady()` to defer the check  on "itemReady" with a Promise, and is cancelled by calling `cancelAutostartFallbackOnItemReady()`.

### Why is this Pull Request needed?

The player should autostart after calling `load` when autostart is enabled. Typically we track the reason for these plays as autostart plays in the "playAttempt" and "play" events' "playReason".

When `play()` is called after load, before the new item is auto-started, we want to track the "playAttempt" and "play" events' "playReason" as the the reason given by the call to play(), not autostart.

This allows us to provide accurate play reasons when UI elements load and play new content with autostart enabled, without turning off autostart which would change the expected behavior when `play()` is not called following a call to `load()`.

#### Addresses Issue(s):

JW7-4465

### CODE REVIEW

**Comment by dannyfinks** [comment]
test this please

**Comment by robwalch** [comment]
@dannyfinks @mikesalvia 
```
Running "squash" task
/var/lib/jenkins-slave/workspace/jw7-commercial/test/squash/feature-stats.js:52
            if (err) throw err;
                     ^

Error: ENOENT: no such file or directory, open '/var/lib/jenkins-slave/workspace/jw7-commercial/test/cucumber/features/analytics_play_reason_autostart.feature'
    at Error (native)
Warning: Command failed: node parse-squashed-files.js Use --force to continue.

Aborted due to warnings.
```

**Comment by dannyfinks** [comment]
@mikesalvia @robwalch its missing the directory in front of it - I'm fixing it now

**Comment by robwalch** [comment]
Thanks @dannyfinks. I restarted the PR build.

**Comment by robwalch** [comment]
I got this one. Jenkins expects a grunt subtask for `karma:jenkins` that wasn't added to v7.12.x.

File: src/js/controller/controller.js [hunk 1]
```diff
@@ -281,6 +282,28 @@ define([
...
+            }
+
+            function cancelAutostartFallbackOnItemReady() {
+                checkAutoStartLastContext.bail = true;
+                _model.off('itemReady', checkAutoStartLast);
+            }
+
+            function checkAutoStartLast() {
+                // Use promise as setImmediate() to allow synchonous calls to load() and play() set the playReason
+                Promise.resolve().then(() => {
```
Comment (egreaves) [inline src/js/controller/controller.js]: Interesting. I'd long wondered if there was a browser equivalent of `setImmediate`.  Seems only IE/Edge will ever implement it.

**Pull request merged**